terraform/eval_diff.go

```diff
@@ -238,6 +238,38 @@ func (n *EvalDiffTainted) Eval(ctx EvalContext) (interface{}, error) {
 	return nil, nil
 }
 
+// EvalFilterDiff is an EvalNode implementation that filters the diff
+// according to some filter.
```

Comment:
Going over this again, I'm curious as to why there's even a boolean parameter. It's only used in one place:

So it could have no params and just be called `EvalStripEverythingButDestroy`?

I guess if you're trying to make these Eval nodes more like generic operations, we could do the function param thing?

```go
type EvalFilterDiff struct {
  Filter func(**InstanceDiff) (**InstanceDiff, error)
}

// ...

&EvalFilterDiff{
  Diff:    &diffApply,
  Output:  &diffApply,
  Filter: func(n **InstanceDiff) (* InstanceDiff, error) {
    input := *n.Diff
    result := new(InstanceDiff)
    if input.Destroy || input.RequiresNew() {
      result.Destroy = true
    }
    return result, nil
  }
},

```